### DISCUSSION TITLE
Postgres - 'ON CONFLICT DO NOTHING RETURNING id' statement

### DISCUSSION THREAD

**Discussion by vlauciani** [laravel/framework#39487] [Ideas]
I'm using Postgres and I need to submit a query like:

```
insert into mytable (field1, field2) values (value1, value2) ON CONFLICT DO NOTHING RETURNING id
```

I tried to use `insertOrIgnore()` method but It build the query with only `ON CONFLICT DO NOTHING`; I tried to use `insertGetId()` method but It build the query with only `RETURNING id`.

Is there a way to "chain" the two methods?

I also tried to build a raw insert:
```
DB::insert('insert into mytable (field1, field2) values (?, ?) ON CONFLICT DO NOTHING returning "id"', [
                $value1,
                $value2
            ]);
```
but It doesn't return the `id` when the record is missing; It returns only `true` or `false`.

---

Could It be possibile to implement a new method like:
```
    public function compileInsertOrIgnoreAndGetId(Builder $query, $values, $sequence)
    {
        return $this->compileInsert($query, $values).' on conflict do nothing returning '.$this->wrap($sequence ?: 'id');
    }
```

into:
- https://github.com/laravel/framework/blob/8.x/src/Illuminate/Database/Query/Grammars/PostgresGrammar.php

Thank you

**Comment by antonkomarev** [comment]
@vlauciani you can try it in next 13.x release:
- https://github.com/laravel/framework/pull/59025
